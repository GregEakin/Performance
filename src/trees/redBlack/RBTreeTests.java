package trees.redBlack;

import org.approvaltests.Approvals;
import org.junit.Test;

import static org.junit.Assert.*;

public class RBTreeTests {

    @Test
    public void LeftRotateRootTest() throws Exception {
        RBTree tree = new RBTree();
        tree.root = RBNodeTests.buildSimpleTree();

        tree.LeftRotate(tree.root);
        Approvals.verify(tree.root.dumpTree(""));
    }

    @Test
    public void LeftRotateNodeTest() throws Exception {
        RBTree tree = new RBTree();
        tree.root = new RBNode();
        tree.root.payload = "root";
        tree.root.right = RBNodeTests.buildSimpleTree();
        tree.root.right.parent = tree.root;

        tree.LeftRotate(tree.root.right);
        Approvals.verify(tree.root.dumpTree(""));
    }
}
