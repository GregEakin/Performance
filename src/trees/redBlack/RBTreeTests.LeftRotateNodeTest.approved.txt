 -- root
    -- y
          -- x
                -- α
                -- β
          -- γ
